src/__tests__/patients/care-goals/CareGoalForm.test.tsx

```diff
@@ -0,0 +1,292 @@
+    const wrapper = setup(false, false)
+
+    act(() => {
+      const noteInput = wrapper.findWhere((w) => w.prop('name') === 'note')
+      const onChange = noteInput.prop('onChange') as any
+      onChange({ currentTarget: { value: expectedNote } })
+    })
+
+    expect(onCareGoalChangeSpy).toHaveBeenCalledWith({ note: expectedNote })
+  })
+
+  it('should render all the forms fields disabled if the form is disabled', () => {
+    const wrapper = setup(true)
+
+    const description = wrapper.findWhere((w) => w.prop('name') === 'description')
+    const priority = wrapper.findWhere((w) => w.prop('name') === 'priority')
+    const status = wrapper.findWhere((w) => w.prop('name') === 'status')
+    const achievementStatus = wrapper.findWhere((w) => w.prop('name') === 'achievementStatus')
+    const startDate = wrapper.findWhere((w) => w.prop('name') === 'startDate')
+    const dueDate = wrapper.findWhere((w) => w.prop('name') === 'dueDate')
+    const note = wrapper.findWhere((w) => w.prop('name') === 'note')
+
+    expect(description.prop('isEditable')).toBeFalsy()
+    expect(priority.prop('isEditable')).toBeFalsy()
+    expect(status.prop('isEditable')).toBeFalsy()
+    expect(achievementStatus.prop('isEditable')).toBeFalsy()
+    expect(startDate.prop('isEditable')).toBeFalsy()
+    expect(dueDate.prop('isEditable')).toBeFalsy()
+    expect(note.prop('isEditable')).toBeFalsy()
+  })
+
+  it('should render the forms field in an error state', () => {
+    const expectedError = {
+      message: 'some error message',
+      description: 'some description error',
+      status: 'some status error',
+      achievementStatus: 'some achievement status error',
+      priority: 'some priority error',
+      startDate: 'some start date error',
+      dueDate: 'some due date error',
+      note: 'some note error',
+    }
+
+    const wrapper = setup(false, false, expectedError)
+
+    const alert = wrapper.find(Alert)
+    const descriptionInput = wrapper.findWhere((w) => w.prop('name') === 'description')
+    const prioritySelector = wrapper.findWhere((w) => w.prop('name') === 'priority')
+    const statusSelector = wrapper.findWhere((w) => w.prop('name') === 'status')
+    const achievementStatusSelector = wrapper.findWhere(
+      (w) => w.prop('name') === 'achievementStatus',
+    )
+    const startDatePicker = wrapper.findWhere((w) => w.prop('name') === 'startDate')
+    const dueDatePicker = wrapper.findWhere((w) => w.prop('name') === 'dueDate')
+    const noteInput = wrapper.findWhere((w) => w.prop('name') === 'note')
+
+    expect(alert).toHaveLength(1)
+    expect(alert.prop('message')).toEqual(expectedError.message)
+
+    expect(descriptionInput.prop('isInvalid')).toBeTruthy()
+    expect(descriptionInput.prop('feedback')).toEqual(expectedError.description)
+
+    expect(prioritySelector.prop('isInvalid')).toBeTruthy()
+    // expect(prioritySelector.prop('feedback')).toEqual(expectedError.priority)
```

Comment:
I think this commented out code should be tested. We should un-comment it and make sure the tests pass.

Reply:
I think there is no feedback prop on SelectWithLabelFormGroup component. Also has a coment "_todo: add feedback in next round_"